lib/src/license.dart

```diff
@@ -53,22 +55,31 @@ abstract class LicenseNames {
   static const String unknown = 'unknown';
 }
 
-Future<List<License>> detectLicensesInDir(String baseDir) async {
-  var list = await new Directory(baseDir).list().toList();
-  final File licenseFile = list.firstWhere(_isLicenseFile, orElse: () => null);
-  if (licenseFile == null) {
-    return [];
-  }
-  return [await detectLicenseInFile(licenseFile)];
+Future<List<LicenseFile>> detectLicensesInDir(String baseDir) async {
+  final rootFiles = await new Directory(baseDir).list().toList();
+  final licenseCandidates =
+      rootFiles.where((fse) => fse is File).where(_isLicenseFile).toList();
+  if (licenseCandidates.isEmpty) return [];
+
+  final licenses = await Future.wait(licenseCandidates.map(
```

Comment:
Assuming this list is not dozens long, I guess it's fine...

Reply:
shall we limit it to 10?

Reply:
Something like that...